src/include/denc.h

```diff
@@ -497,6 +497,22 @@ inline void denc_varint_lowz(T& v, ceph::buffer::ptr::const_iterator& p)
   v = i;
 }
 
+// for standard library which does not fully support C++20
+#ifndef __cpp_lib_bit_cast
```

Comment:
cool, i see that this is from https://en.cppreference.com/w/cpp/utility/feature_test so we should expect it to be portable across standard libaries/compilers